### DISCUSSION TITLE
Storybook. Tailwind, Create React App Compatibility

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#5012] [Help]
I am trying to configure my application to load tailwind styles in Storybook but I am having no success so far.  I am using ` "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.2.4"` to allow for compatibility with Create React App.  Everything works fine in the react application, but the problems occur in storybook.

main.js
```
const path = require("path");

module.exports = {
  stories: ["../src/**/*.stories.mdx", "../src/**/*.stories.@(js|jsx|ts|tsx)"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/preset-create-react-app",
    "@storybook/addon-postcss",
  ],
  webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.css$/,
      use: [
        {
          loader: "postcss-loader",
          options: {
            ident: "postcss",
            plugins: [require("tailwindcss"), require("autoprefixer")],
          },
        },
      ],
      include: path.resolve(__dirname, "../"),
    });
    return config;
  },
};
```

preview.js
```
import "../src/index.css";
export const parameters = {
  actions: { argTypesRegex: "^on[A-Z].*" },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
};
```

I get the following error when attempting to run storybook.  I get this if I attempt to import any css file into any of my Stories.

```
ModuleBuildError: Module build failed (from ./node_modules/react-scripts/node_modules/postcss-loader/src/index.js):
SyntaxError

(1:1) Unknown word

> 1 | var api = require("!../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js");
    | ^
  2 |             var content = require("!!../node_modules/@storybook/addon-postcss/node_modules/css-loader/dist/cjs.js!../node_modules/@storybook/addon-postcss/node_modules/postcss-loader/dist/cjs.js!../node_modules/react-scripts/node_modules/postcss-loader/src/index.js??postcss!./index.css");
  3 | 
```

Are there are examples available that can show me how to configure this project?  I have tried everything I have been able to find online.

**Comment by dance2die** [comment]
I had a similar issue with [TSDX](https://tsdx.io/) and my workaround was to use the storybook's postcss addon.  
Might not work as my project doesn't use the compat version. (try to use postcss 7 for the implementation).

https://github.com/downwindcss/easings/blob/main/.storybook/main.js#L7-L14

```js
module.exports = {
  stories: ['...'],
  addons: [
   ...
    {
      name: '@storybook/addon-postcss',
      options: {
        postcssLoaderOptions: {
          implementation: require('postcss'),
        },
      },
    },
  ],
  webpackFinal: async config => {
   ...
  },
};
```

**Reply by ghost** [reply]
> I had a similar issue with [TSDX](https://tsdx.io/) and my workaround was to use the storybook's postcss addon.
> Might not work as my project doesn't use the compat version. (try to use postcss 7 for the implementation).
> 
> https://github.com/downwindcss/easings/blob/main/.storybook/main.js#L7-L14
> 
> ```js
> module.exports = {
>   stories: ['...'],
>   addons: [
>    ...
>     {
>       name: '@storybook/addon-postcss',
>       options: {
>         postcssLoaderOptions: {
>           implementation: require('postcss'),
>         },
>       },
>     },
>   ],
>   webpackFinal: async config => {
>    ...
>   },
> };
> ```

Yes I have tried this configuration as well.  I am guessing that I need to continue to use the compat version because I am using Create React App.  I still get the same error. even when configuring like this.

Thanks!

**Comment by paulwongx** [comment]
Hey I finally got it to work. Please see my SO link below and let me know if it works for you!

https://stackoverflow.com/questions/71329335/how-to-setup-storybook-with-tailwindcss-reactjs-and-typescript
